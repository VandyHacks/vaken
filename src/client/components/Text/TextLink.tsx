import styled from 'styled-components';
import { Link, LinkProps } from 'react-router-dom';

export interface Props extends LinkProps {
	color?: string;
	fontFamily?: string;
	fontSize?: string;
	fontWeight?: number;
	glowColor?: string;
	textDecoration?: string;
}

const TextLink = styled(Link)`
	font-family: ${(props: Props) => props.fontFamily || "'Roboto Condensed'"};
	font-size: ${(props: Props) => props.fontSize || '1rem'};
	font-weight: ${(props: Props) => props.fontWeight || 500};
	color: ${(props: Props) => props.color || 'black'};
	text-decoration: ${(props: Props) => props.textDecoration || 'none'};
	&:hover,
	&:focus,
	&:active {
		text-shadow: 0px 0px 20px ${({ glowColor }: Props) => glowColor || 'RGBA(255, 255, 255, 0.67)'};
	}
`;

export default TextLink;
