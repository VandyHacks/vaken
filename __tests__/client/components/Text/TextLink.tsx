import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import TextLink from '../../../../src/client/components/Text/TextLink';
describe('Test TextLink', () => {
	it('renders correctly', async () => {
		const component = renderer
			.create(
				<MemoryRouter>
					<TextLink to="/" />
				</MemoryRouter>
			)
			.toJSON();
		expect(component).toMatchSnapshot();
	});
});
