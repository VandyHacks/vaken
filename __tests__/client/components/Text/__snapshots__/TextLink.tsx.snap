// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextLink renders correctly 1`] = `
<a
  className="sc-bdVaJa cXunfN"
  href="/"
  onClick={[Function]}
/>
`;
